Implement new command 'pip index versions' (#8978)
pypa/pip: news/7975.feature.rst

> Add new subcommand "index" used to interact with indecies.
> Also add an option "versions" for pip index, used to list of available versions
> of a package.

Reviewer: ```suggestion
Add an experimental ``pip index`` subcommand, for interacting with package indexes.
```
